File: OC.UI/src/app/blci/api/branches/church-program-session-visitor.ts
import { ChurchProgramSession } from "./church-program-session";

export interface ChurchProgramSessionVisitor {
    id?: number;
    churchProgramSessionId?: number;
    name?: string;
    mobileNumber?: string;
    email?: string;
    address?: string;
    dateVisited?: Date;
    dateToVisit?: Date;
    followedUp?: boolean;
    comments?: string;
    churchProgramSession?: ChurchProgramSession;
}

File: OC.Domain/Models/Branches/ChurchProgramSessionVisitor.cs
using OC.Domain.Interfaces;

namespace OC.Domain.Models.Branches
{
    public class ChurchProgramSessionVisitor : AuditableEntity, IEntityBase
    {
        public long Id { get; set; }
        public long ChurchProgramSessionId { get; set; }
        // Add other fields
        public string? Name { get; set; }
        public string? MobileNumber { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }
        public DateTime? DateVisited { get; set; }
        public DateTime? DateToVisit { get; set;}
        public bool? FollowedUp { get; set; }
        public string? Comments { get; set; }

        // Relationships
        public virtual ChurchProgramSession? ChurchProgramSession{ get; set; }
    }
}
